File: Odd Even Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
   public:
    ListNode* oddEvenList(ListNode* head) {
        if (!head) return head;
        ListNode* E = head;
        ListNode* O = head->next;
        ListNode* t = O;
        if (!O) return head;
        while (1) {
            E->next = O->next;
            E = E->next;
            if (!E) break;
            O->next = E->next;
            O = O->next;
            if (!O) break;
        }
        ListNode* temp = head;
        for (; temp->next; temp = temp->next)
            ;
        temp->next = t;
        return head;
    }
};

File: odd-even-linked-list/odd-even-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return head;
        E = head;
        O = head.next;
        t = O;
        if not O:
            return head;
        while True:
            E.next = O.next;
            E = E.next;
            if not E:
                break;
            O.next = E.next;
            O = O.next;
            if not O:
                break;
        
        temp = head;
        while temp.next:
            temp = temp.next
    
        temp.next = t;
        return head;